sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_deployment/online/data_collector_schema.py

```diff
@@ -7,27 +7,34 @@
 import logging
 from typing import Any
 
-from marshmallow import fields, post_load
+from marshmallow import fields, post_load, validates, ValidationError
 
 from azure.ai.ml._schema import NestedField, PatchedSchemaMeta, StringTransformedEnum
 from azure.ai.ml._schema._deployment.online.destination_schema import DestinationSchema
 from azure.ai.ml._schema._deployment.online.request_logging_schema import RequestLoggingSchema
-from azure.ai.ml._schema._deployment.online.sampling_strategy_schema import SamplingStrategySchema
+from azure.ai.ml._schema._deployment.online.deployment_collection_schema import DeploymentCollectionSchema
+
 from azure.ai.ml.constants._common import RollingRate
 
 module_logger = logging.getLogger(__name__)
 
 
 class DataCollectorSchema(metaclass=PatchedSchemaMeta):
-    enabled = fields.Bool()
+    collections = fields.Mapping(fields.Str, NestedField(DeploymentCollectionSchema))
     rolling_rate = StringTransformedEnum(
         required=False,
-        allowed_values=[RollingRate.HOUR, RollingRate.YEAR, RollingRate.MONTH, RollingRate.DAY, RollingRate.MINUTE],
+        allowed_values=[ RollingRate.MINUTE, RollingRate.DAY, RollingRate.HOUR],
     )
     destination = NestedField(DestinationSchema)
-    sampling_strategy = NestedField(SamplingStrategySchema)
+    sampling_rate = fields.Float()
     request_logging = NestedField(RequestLoggingSchema)
 
+    # pylint: disable=unused-argument,no-self-use
+    @validates("sampling_rate")
+    def validate_sampling_rate(self, value, **kwargs):
+        if value > 1.0 or value < 0.0:
+            raise ValidationError("Sampling rate must be an number between 0.0-1.0")
```

Comment:
```suggestion
            raise ValidationError("Sampling rate must be a number between 0.0-1.0")
```

Reply:
Tho the mathematician in me prefers `in range [0.0, 1.0]` so it's clear about the bounds being inclusive.

Reply:
In this case the bounds are exclusive, but I will still make the range change with parentheses